app/experimenter/experiments/tests/factories/nimbus.py

```diff
@@ -256,11 +259,15 @@ def create_with_lifecycle(cls, lifecycle, **kwargs):
             if experiment.has_filter(experiment.Filters.SHOULD_ALLOCATE_BUCKETS):
                 experiment.allocate_bucket_range()
 
-            generate_nimbus_changelog(
+            change = generate_nimbus_changelog(
                 experiment,
                 experiment.owner,
                 f"set lifecycle {lifecycle} state {state}",
             )
+            if with_random_timespan:
+                change.changed_on = now
+                change.save()
+                now += datetime.timedelta(days=random.randint(5, 20))
```

Comment:
This is slightly hacky. But, I had an earlier attempt that modified `generate_nimbus_changelog` to add an optional `changed_on` parameter that seemed to add more complexity than necessary IMO

Reply:
Oh oh! Thank you! I've actually been craving something like this lately, so thank you for hopping in and doing it.  This is a totally fine way to do it, but one thing I've been wanting lately is the ability to explicitly set like 'make this this exact many days in the past' and I was thinking about sane ways to do it, one is you could pass in a negative delta paramter, but that's a bit meh, but what about passing in a start date datetime object, which would basically take the place of your 'now' initializer, so if you don't pass it in, it just starts at now now, and if you do, it starts in the past (or future if you needed it to I guess).  And then I think combined with this you could basically cover all the cases I've been thinking of.  What do you think?